pkg/recordset/recordset.go

```diff
@@ -225,6 +225,25 @@ func (m *Manager) createMissingTargetStacks(sourceStacks, targetStacks []string)
 }
 
 func (m *Manager) deleteOrphanTargetStacks(sourceStacks, targetStacks []string) error {
+	for _, target := range targetStacks {
+		found := false
+		targetClusterName := extractClusterName(target)
+		for _, source := range sourceStacks {
+			sourceClusterName := extractClusterName(source)
+			if sourceClusterName == targetClusterName {
+				found = true
+				break
+			}
+		}
+		if !found {
+			err := m.deleteTargetStack(target)
+			if err != nil {
+				m.logger.Log("level", "debug", "message", fmt.Sprintf("%q stack deleted", target))
+			} else {
+				m.logger.Log("level", "debug", "message", fmt.Sprintf("failed to delete %q stack", target))
```

Comment:
Yes I also think logs are in wrong blocks :)
Also when logging error, I think it would be good to log `stack` key.

Reply:
Good catch! Fixed and added stack, PTAL.